query/resource_selector.go

```diff
@@ -224,24 +247,62 @@ func SetResourceSelectorClause(ctx context.Context, resourceSelector types.Resou
 		}
 	}
 
-	if resourceSelector.Search != "" {
-		var prefixQueries []*gorm.DB
-		if resourceSelector.Name == "" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("name ILIKE ?", resourceSelector.Search+"%"))
-		}
-		if resourceSelector.TagSelector == "" && table == "config_items" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(tags) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
+	if resourceSelector.Functions.ComponentConfigTraversal != nil {
+		args := resourceSelector.Functions.ComponentConfigTraversal
+		if table == "components" {
+			query = query.Where("id IN (SELECT id from lookup_component_config_id_related_components(?, ?, ?))", args.ComponentID, args.Direction, args.Types)
 		}
-		if resourceSelector.LabelSelector == "" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(labels) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
-		}
-
-		query = OrQueries(query, prefixQueries...)
 	}
 
 	return query, nil
 }
 
+func setSearchQueryParams(rs *types.ResourceSelector) {
+	if rs.Search == "" {
+		return
+	}
+
+	queries := strings.Split(rs.Search, " ")
+	for _, q := range queries {
+		items := strings.Split(q, "=")
+		if len(items) != 2 {
+			continue
+		}
+
+		switch items[0] {
+		case "component_config_traverse":
+			// search: component_config_traverse=72143d48-da4a-477f-bac1-1e9decf188a6,outgoing,Kubernetes::Pod,Kubernetes::Node
+			// Args should be componentID, direction and types (compID,direction,type1,type2,type3)
+			args := strings.SplitN(items[1], ",", 3)
+			if len(args) == 3 {
+				rs.Functions.ComponentConfigTraversal = &types.ComponentConfigTraversalArgs{
+					ComponentID: args[0],
+					Direction:   args[1],
+					Types:       types.Items(strings.Split(args[2], ",")),
```

Comment:
why not just use the main selector types for this, 

i.e. 
```
component_config_traverse=72143d48-da4a-477f-bac1-1e9decf188a6. 
 direction=outgoing type=Kubernetes::Pod,Kubernetes::Node
```

Reply:
Yeah, makes sense